src/chordSymbol.js

```diff
@@ -0,0 +1,593 @@
+        rv += -1;
+      }
+    }
+    return rv;
+  }
+
+  // ### getSymbolBlock
+  // ChordSymbol allows multiple blocks so we can mix glyphs and font text.
+  // Each block can have its own vertical orientation
+  getSymbolBlock(parameters) {
+    parameters = parameters == null ? {} : parameters;
+    const symbolType = (parameters.symbolType ? parameters.symbolType : ChordSymbol.SymbolTypes.TEXT);
+    const text = parameters.text ? parameters.text : '';
+    const symbolModifier = parameters.symbolModifier ? parameters.symbolModifier : ChordSymbol.SymbolModifiers.NONE;
+    const x_offset = 0;
+    const y_offset = 0;
+
+    const rv = {
+      text, symbolType, symbolModifier, x_offset, y_offset
+    };
+
+    rv.width = 0;
+    if (symbolType === ChordSymbol.SymbolTypes.GLYPH && typeof(parameters.glyph) === 'string') {
+      const glyphArgs = ChordSymbol.GLYPHS[parameters.glyph];
+      let glyphPoints = 20 * glyphArgs.scale;
+      // super and subscript are smaller
+      if (symbolModifier !== ChordSymbol.SymbolModifiers.NONE) {
+        glyphPoints = glyphPoints / 1.3;
+      }
+      rv.glyph = new Glyph(glyphArgs.code, glyphPoints, { category: 'chordSymbol' });
+      rv.width = (rv.glyph.getMetrics().width * 3) / 2;
+      rv.y_offset = glyphArgs.y_offset;
+    } else if (symbolType === ChordSymbol.SymbolTypes.TEXT) {
+      rv.width = Flow.textWidth(text);
+    } else if (symbolType === ChordSymbol.SymbolTypes.LINE) {
+      rv.width = parameters.width;
+    }
+
+    return rv;
+  }
+
+  addSymbolBlock(parameters) {
+    this.symbolBlocks.push(this.getSymbolBlock(parameters));
+    return this;
+  }
+
+  // ### Convenience functions follow that let you create different types of
+  // chord symbol parts easily
+  addText(text, parameters) {
+    parameters = parameters == null ? {} : parameters;
+    parameters.text = text;
+    parameters.symbolType = ChordSymbol.SymbolTypes.TEXT;
+    return this.addSymbolBlock(parameters);
+  }
+
+  addSuperText(text) {
+    const symbolType = ChordSymbol.SymbolTypes.TEXT;
+    const symbolModifier = ChordSymbol.SymbolModifiers.SUPERSCRIPT;
+    return this.addSymbolBlock({ text, symbolType, symbolModifier });
+  }
+
+  addSubText(text) {
+    const symbolType = ChordSymbol.SymbolTypes.TEXT;
+    const symbolModifier = ChordSymbol.SymbolModifiers.SUBSCRIPT;
+    return this.addSymbolBlock({ text, symbolType, symbolModifier });
+  }
+
+  addSuperGlyph(glyph) {
+    const symbolType = ChordSymbol.SymbolTypes.GLYPH;
+    const symbolModifier = ChordSymbol.SymbolModifiers.SUPERSCRIPT;
+    return this.addSymbolBlock({ glyph, symbolType, symbolModifier });
+  }
+
+  addGlyph(glyph, parameters) {
+    parameters = parameters == null ? {} : parameters;
+    parameters.glyph = glyph;
+    parameters.symbolType = ChordSymbol.SymbolTypes.GLYPH;
+    return this.addSymbolBlock(parameters);
+  }
+
+  addGlyphOrText(text, parameters) {
```

Comment:
Not thrilled with this function name. What exactly is the purpose of this?

Reply:
Here is the description, feel free to suggest a better name:
```
  // ### addGlyphOrText
  // Add a glyph for each character in 'text'.  If the glyph is not
  // available, use text from the font.  E.g.:
  // `addGlyphOrText("(+5#11)")`
  // will use text for the '5' and '11', and glyphs for everything else.
```

Reply:
Okay, this is fine -- thanks.